packages/constants/src/lib/errors.ts

```diff
@@ -171,3 +190,103 @@ export const LIT_ERROR = {
 export const LIT_ERROR_CODE = {
   NODE_NOT_AUTHORIZED: 'NodeNotAuthorized',
 };
+
+export abstract class LitError extends VError {
+  protected constructor(
+    options: Error | Options,
+    message: string,
+    ...params: any[]
+  ) {
+    super(options, message, ...params);
+  }
+}
+
+type LitErrorConstructor = new (
+  options: Error | Options,
+  message: string,
+  ...params: any[]
+) => LitError;
+
+function createErrorClass({
+  name,
+  code,
+  kind,
+}: {
+  name: string;
+  code: string;
+  kind: string;
+}): LitErrorConstructor {
+  return class extends LitError {
+    // VError has optional options parameter, but we make it required so thrower remembers to pass all the useful info
+    constructor(options: Error | Options, message: string, ...params: any[]) {
+      if (options instanceof Error) {
+        options = {
+          cause: options,
+        };
+      }
+
+      // If the cause is not an Error, wrap it in one
+      if (!(options.cause instanceof Error)) {
+        options.cause = new Error(options.cause);
```

Comment:
Will wrapping here have implications on the stack trace? Since this class is non named and we are creating a new instance of `Error` in scope of a function call we might want to take the stack trace from the `options` if it is of type `Error`

Reply:
This is added because `cause` has to be an `Error` or the `VError` constructor will throw itself, therefore we wrap to comply with it.
When that happens thrower threw something else and we do not have a stacktrace originating from the thrower and hide.

Reply:
I see now, since we are in scope of a constructor  the stack trace is going is going to be from the scope of the constructor call in a way preserve the stack context.